chain-crypto/src/key.rs

```diff
@@ -150,6 +150,12 @@ impl<A: AsymmetricPublicKey> AsRef<[u8]> for PublicKey<A> {
     }
 }
 
+impl<A: AsymmetricKey> AsRef<[u8]> for SecretKey<A> {
```

Comment:
This is unrelated to the rest of this PR but it's just a functionality I wanted that way you can implement `as_bytes` for private keys using

```rust
pub fn as_bytes(&self) -> Vec<u8> {
        self.0.as_ref().to_vec()
    }
```